src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -963,6 +966,18 @@ private ISet<string> CalculateFilesToExcludeInPack(IPackTaskRequest<IMSBuildItem
                         }
                     }
 
+                    if (packageDependency.NoWarn.Count > 0)
+                    {
+                        StringBuilder sb = new StringBuilder();
+
+                        foreach (NuGetLogCode nuGetLogCode in packageDependency.NoWarn)
+                        {
+                            sb.Append(" " + nuGetLogCode.ToString());
+                        }
+
+                        packageBuilder.Properties[NoWarn] = sb.ToString().Trim();
```

Comment:
You should use a semicolon delimiter instead of space and [`string.Join()`](https://docs.microsoft.com/en-us/dotnet/api/system.string.join) since its [highly optimized](https://source.dot.net/#System.Private.CoreLib/String.Manipulation.cs,581) to do just this:

```C#
packageBuilder.Properties[NoWarn] = string.Join(';', packageDependency.NoWarn);
```

Reply:
Wouldn't this NoWarn *everything*, and not just the package in question?

Reply:
I don't follow what you mean. What is the everything?

Reply:
Say you have 2 PRs. 

```xml

<PackageReference Include="A" Version="1.0.0" NoWarn="NU1567" />
<PackageReference Include="B" Version="1.0.0" />
```

Say B was raising NU1567, it would be no warned, but it shouldn't be. 

Only A raising NU1567 should be no warned.